system/CLI/GeneratorTrait.php

```diff
@@ -0,0 +1,318 @@
+		{
+			CLI::write(lang('CLI.generator.usingCINamespace'), 'yellow');
+			CLI::newLine();
+
+			// @codeCoverageIgnoreStart
+			if (CLI::prompt('Are you sure you want to continue?', ['y', 'n'], 'required') === 'n')
+			{
+				CLI::newLine();
+				CLI::write(lang('CLI.cancelOperation'), 'light_yellow');
+				CLI::newLine();
+
+				return;
+			}
+			// @codeCoverageIgnoreEnd
+
+			CLI::newLine();
+		}
+
+		// Get the fully qualified class name from the input.
+		$class = $this->qualifyClassName();
+
+		// Get the file path from class name.
+		$path = $this->buildPath($class);
+
+		// Check if path is empty.
+		if (empty($path))
+		{
+			return;
+		}
+
+		// Overwriting files unknowingly is a serious annoyance, So we'll check if
+		// we are duplicating things, If 'force' option is not supplied, we bail.
+		if (! $this->getOption('force') && file_exists($path))
+		{
+			CLI::write(lang('CLI.generator.fileExist', [CLI::color(clean_path($path), 'red')]));
+			CLI::newLine();
+
+			return;
+		}
+
+		// Check if the directory to save the file is existing.
+		$dir = dirname($path);
+
+		if (! is_dir($dir))
+		{
+			mkdir($dir, 0755, true);
+		}
+
+		helper('filesystem');
+		
+		// Build the class based on the details we have, We'll be getting our file
+		// contents from the template, and then we'll do the necessary replacements.
+		if (! write_file($path, $this->buildContent($class)))
+		{
+			CLI::write(lang('CLI.generator.fileError', [clean_path($path)]), 'red');
+			CLI::newLine();
+
+			return;
+		}
+
+		if ($this->getOption('force') && file_exists($path))
+		{
+			CLI::write(lang('CLI.generator.fileOverwrite', [CLI::color(clean_path($path), 'yellow')]));
+			CLI::newLine();
+
+			return;
+		}
+
+		CLI::write(lang('CLI.generator.fileCreate', [CLI::color(clean_path($path), 'light_green')]));
+		CLI::newLine();
+	}
+
+	/**
+	 * Prepare options and do the necessary replacements.
+	 *
+	 * @param string $class
+	 *
+	 * @return string
+	 */
+	protected function prepare(string $class): string
+	{
+		return $this->parseTemplate($class);
+	}
+
+	/**
+	 * Change file basename before saving.
+	 * 
+	 * Useful for components where the file name has a date.
+	 *
+	 * @param string $filename
+	 *
+	 * @return string
+	 */
+	protected function basename(string $filename): string
+	{
+		return basename($filename);
+	}
+
+	/**
+	 * Parses the class name and checks if it is already qualified.
+	 *
+	 * @return string
+	 */
+	private function qualifyClassName(): string
+	{
+		// Gets the class name from input.
+		$class = $this->params[0] ?? CLI::getSegment(2) ?? '';
+
+		if (empty($class))
+		{
+			$class = CLI::prompt(lang('CLI.generator.className'), null, 'required'); // @codeCoverageIgnore
+			CLI::newLine();
+		}
+
+		helper('inflector');
+
+		$component = strtolower(singular($this->component));
+		$class     = strtolower($class);
+		$class     = strpos($class, $component) ? str_replace($component, ucfirst($component), $class) : $class;
```

Comment:
This needed because without it, if someone append component name to class name `make:controller blogcontroller` it will return `Blogcontroller`, i used this trick to check if the class name includes component name then capitalize component name `BlogController`